modules/social_features/social_group/src/Plugin/Field/FieldWidget/SocialGroupSelectorWidget.php

```diff
@@ -79,13 +83,43 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   protected function getOptions(FieldableEntityInterface $entity) {
     if (!isset($this->options)) {
+
+      // Get the bundle fron the node.
+      $entity_type = '';
+      if ($entity->getEntityTypeId() == 'node') {
+        $entity_type = $entity->bundle();
+      }
+
       $account = $entity->getOwner();
       // Limit the settable options for the current user account.
       $options = $this->fieldDefinition
         ->getFieldStorageDefinition()
         ->getOptionsProvider($this->column, $entity)
         ->getSettableOptions($account);
 
+      // Check for each group type if the content type is installed.
+      foreach ($options as $key => $optgroup) {
+        // Groups are in the array below.
+        if (is_array($optgroup)) {
+          // Loop through the groups.
+          foreach ($optgroup as $gid => $title) {
+            // If the group exists.
+            if ($group = Group::load($gid)) {
+              // Load all installed plugins for this group type.
+              $plugin_ids = $this->pluginManager->getInstalledIds($group->getGroupType());
+              // If the bundle is not installed,
+              // then unset the entire optiongroup (=group type).
+              if (!in_array('group_node:' . $entity_type, $plugin_ids)) {
```

Comment:
@jochemvn What if $entity_type = ''?

Reply:
@jaapjan  then it will not be in the array and nothing will happen, since there's nothing to uninstall

Reply:
Yeahhhh.. maybe I formulated my question wrongly. Shouldn't we check if entity_type === 'node' before executing that code?

Reply:
Yes that would be a good addition I think. Is it even possible to add this fields to other entities? I guess yes right? I mean eg Taxonomies are also fieldable... 🤔

Reply:
Yes in theory you can put every entity in a group

Reply:
Marking as 'needs work' for now then :-)

Reply:
This is what social_group does:
```
  $entity_types = ['node', 'user'];
  if (in_array($entity_type->id(), $entity_types)) {
```
So I'll limit it to nodes and users